### PR TITLE
Fix `WithSingleton` and `WithSingletonProxy` API

### LINKED ISSUE #509: WithSingleton extension method ignores singleton name specified in Akka HOCON
When we create singletons without using Akka.Hosting extensions, the actual singleton actor has path **//user/my_singleton_name/singleton**, i.e. if the name for a singleton manager actor is "my_singleton_name", then the singleton instance is created with "singleton" name, as specified in HOCON (akka.cluster.singleton.singleton-name). And this is correct behavior. However, when using Akka.Hosting WithSingleton extension, the singleton actor path becomes **//user/my_singleton_name/my_singleton_name**.

I believe the name gets overridden in this code (AkkaClusterHostingExtensions.cs):

```
                options ??= new ClusterSingletonOptions();
                var clusterSingletonManagerSettings =
                    ClusterSingletonManagerSettings.Create(system).WithSingletonName(singletonName);

```

First ClusterSingletonManagerSettings.Create(system) creates settings from akka.cluster.singleton config, but then the call to WithSingletonName(singletonName) overrides the value provided in akka.cluster.singleton.singleton-name, so the singleton path contains  the singleton basename (like my_singleton_name) twice instead of the base name with "/singleton" suffix.

### PR DESCRIPTION
**Pull request by Arkatufus**
Fixes #509

## Changes

* Allow for users to specify the singleton manager actor name

## Checklist

For significant changes, please ensure that the following have been completed (delete if not relevant):

* [x] This change follows the [Akka.NET API Compatibility Guidelines](https://getakka.net/community/contributing/api-changes-compatibility.html).
* [x] I have [reviewed my own pull request](https://getakka.net/community/contributing/index.html#review-your-own-pull-requests).
* [x] Changes in public API reviewed, if any.

### CODE REVIEW

File: src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs [hunk 1]
```diff
@@ -160,6 +160,53 @@ public sealed class ClusterSingletonOptions
...
+        
+        /// <summary>
+        /// Should the singleton proxy publish a warning if no singleton actor were found after a period of time
+        /// </summary>
+        public bool? LogSingletonIdentificationFailure { get; set; }
+        
+        /// <summary>
+        /// The period the proxy will wait until it logs a missing singleton warning, defaults to 1 minute
+        /// </summary>
+        public TimeSpan? SingletonIdentificationFailurePeriod { get; set; }
```
Comment (Arkatufus) [inline src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs:177]: Added Akka 1.5.31 missing singleton detection feature

File: src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs [hunk 2]
```diff
@@ -1222,49 +1279,89 @@ public static AkkaConfigurationBuilder WithSingleton<TKey>(
...
+        ///     <list type="bullet">
+        ///         <item>
+        ///             The complete singleton manager actor path name will be
+        ///             <c>akka://MyActorSystem/user/singletonManagerName/singletonName</c>
+        ///         </item>
+        ///         <item>
+        ///             The complete singleton proxy actor path name, if created, will be
+        ///             <c>akka://MyActorSystem/user/singletonName-proxy</c>
+        ///         </item>
+        ///     </list>
```
Comment (Arkatufus) [inline src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs:1300]: Added explanation on what the actor names will be in the XML-DOC

File: src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs [hunk 3]
```diff
@@ -1222,49 +1279,89 @@ public static AkkaConfigurationBuilder WithSingleton<TKey>(
...
+        /// <returns>
+        ///     The same <see cref="AkkaConfigurationBuilder"/> instance originally passed in.
+        /// </returns>
+        public static AkkaConfigurationBuilder WithSingleton<TKey>(
+            this AkkaConfigurationBuilder builder,
+            string singletonManagerName,
+            string singletonName,
+            Func<ActorSystem, IActorRegistry, IDependencyResolver, Props> propsFactory,
+            ClusterSingletonOptions? options = null,
+            bool createProxyToo = true)
```
Comment (Arkatufus) [inline src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs:1336]: New API that allows users to specify the singleton manager actor name.

The resulting singleton path will be `akka://MyActorSystem/user/singletonManagerName/singletonName`

Old APIs are untouched, they will still generate the same singleton actor path `akka://MyActorSystem/user/singletonName/singletonName`

**Review by Arkatufus — COMMENTED** [review]
Self-review

**Review by object — APPROVED** [review]
LGTM

File: src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs (hunk 1 above)
Comment (Aaronontheweb) [reply src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs:177]: :+1:

**Review by Aaronontheweb — APPROVED** [review]
lgtm

**Pull request merged**